client.go

```diff
@@ -63,21 +63,52 @@ func NewClient(apiKey string, options ...Option) *Client {
 }
 
 // Returns `Flags` struct holding all the flags for the current environment.
-func (c *Client) GetEnvironmentFlags() (Flags, error) {
-	if env, ok := c.environment.Load().(*environments.EnvironmentModel); ok {
-		return c.GetEnvironmentFlagsFromDocument(c.ctx, env)
+//
+// If local evaluation is enabled this function will not call the Flagsmith API
+// directly, but instead read the asynchronously updated local environment or
+// use the default flag handler in case it has not yet been updated.
+func (c *Client) GetEnvironmentFlags() (f Flags, err error) {
+	if c.config.localEvaluation {
+		if env, ok := c.environment.Load().(*environments.EnvironmentModel); ok {
```

Comment:
Let's move this to `GetEnvironmentFlagsFromDocument`(make it private) and let it decide if the environment is loaded or not